packages/cxl-ui/src/components/cxl-playbook-accordion.js

```diff
@@ -0,0 +1,163 @@
+import { customElement } from 'lit-element';
+import '@conversionxl/cxl-lumo-styles';
+import '@vaadin/vaadin-accordion';
+import '@vaadin/vaadin-checkbox';
+import { css, registerStyles } from '@vaadin/vaadin-themable-mixin/register-styles.js';
+import { registerGlobalStyles } from '@conversionxl/cxl-lumo-styles/src/utils';
+import cxlAdeftAccordionGlobalStyles from '../styles/global/cxl-playbook-accordion-css.js';
+import { CXLVaadinAccordion } from './cxl-vaadin-accordion';
+
+/**
+ * Allows opening multiple panels simultaneously.
+ * Saves panel state, restores on page load.
+ */
+@customElement('cxl-playbook-accordion')
+export class CxlPlaybookAccordion extends CXLVaadinAccordion {
+  constructor() {
+    super();
+    registerStyles('vaadin-accordion-panel', [CxlPlaybookAccordion._getAccordionPanelStyles()]);
+    registerStyles('vaadin-checkbox', [CxlPlaybookAccordion._getVaadinCheckboxStyles()]);
+  }
+
+  static _getAccordionPanelStyles() {
+    return css`
+      [part='summary'] {
+        box-shadow: 0 4px 12px rgba(36, 38, 40, 0.12);
+        padding: 0;
+      }
+      :host(.emptyContent) [part='toggle'] {
+        display: none;
+      }
+      [part='toggle'] {
+        padding: 12px;
+      }
+    `;
+  }
+
+  static _getVaadinCheckboxStyles() {
+    return css`
+      :host([theme~='custom']) [part='checkbox'] {
+        border: 2px solid !important;
+        width: 18px;
+        height: 18px;
+      }
+      :host([theme~='custom']) [part='label'] {
+        display: none;
+      }
+    `;
+  }
+
+  /**
+   * Global styles.
+   */
+  ready() {
+    super.ready();
+
+    // Define and register a style sheet for the <vaadin-text-field> component
+    registerGlobalStyles(cxlAdeftAccordionGlobalStyles, {
+      moduleId: 'cxl-playbook-accordion-global',
+    });
+
+    this._updateCheckboxesStates();
+  }
+
+  /**
+   *
+   * @param {HTMLCollection} items
+   * @private
+   */
+  _saveAccordionState(items) {
+    const storageId = this.getAttribute('id');
+    // Avoid null key.
+    if (storageId) {
+      const stateItems = [];
+
+      items.forEach((value, key) => {
+        stateItems[key] = items[key].opened;
+      });
+
+      localStorage.setItem(storageId, JSON.stringify(stateItems));
+
+      this._updateCSSAndPanelStateToCheckboxesStates();
+
+      this._dispatchCustomEvent(stateItems);
+
+      this._saveCheckboxesState();
+    }
+  }
+
+  _saveCheckboxesState() {
+    const stateCheckboxes = [];
+    const checkboxes = this.querySelectorAll('vaadin-checkbox');
```

Comment:
This should be class property, for better clarity of how this component operates.

See cxl-marketing-nav for examples.

Reply:
Refactored in: https://github.com/conversionxl/aybolit/pull/87/commits/214d2ee775cafa0fc3bd205f0044ebce9a9700e7
Is this OK?

Reply:
Only way I managed to get it working is this:
https://github.com/conversionxl/aybolit/pull/87/commits/d08d61c4fdbd7b1b1ac8d45876f5e362159ca907
We could also go for computed properties or observers: https://polymer-library.polymer-project.org/3.0/docs/devguide/observers - but I'm not sure it's worth the effort since simple getters are transparent enough.